### ISSUE TITLE
[Suggestion] Ability to Turn Turret Cores Off and On

### ISSUE THREAD

**Issue by Aiwii** [tsunamayo/Starship-EVO#2089]
### Short Description of the Suggestion:
Currently, we can link switches and some other things to the Turret Core but it doesn't do much. I want to suggest that when a switch is hooked up to a Turret Core it turns auto fire off and on. This would prevent the issue of firing at a ship you just want to target and give players more control over what turrets are firing.
...

### Environment
- Starship EVO version: 20w20d
- Graphics Card: Integrated

**Comment by tsunamayo** [comment]
I have to be careful with that, it seems a lot of newcomer are linking stuff a bit randomly, and then their contraption is not working. So I will have to tone down some feature, do them in a different way.
In that case this will be dealt with an advanced "input" gate, an evolution of the sequencer gate, where you can tweak input of your system.

**Comment by ZachZent** [comment]
To add on this, there is the problem that targeting anything will result in your firing at it. This includes places you want to cruise to, ships you want to read their information (heat, etc) landing pads you want to land on, etc.

**Comment by tsunamayo** [comment]
Yes @ZachZent so actually for debugging I deactivate that feature. But turret are not supposed to fire on civilian ship. Which raise the problem of pirate-like play. Not sore on the best way to solve this...
There is different approach, each with pros and cons. I think I will start on something and will likely change it later on if it doesnt work well...

**Comment by ZachZent** [comment]
I think a best option would be some sort of command hotkey/prompt that you can assign in the turret computer. Something you can press to activate/deactivate turrets when flying. One option is using the already in game hotkeys 1-4. This does have the negative that the player wouldn't be able to assign them to anything else. It would be rather awkward to press your hangar door open assigned to hotkey 3 and then broadside a poor civilian ship. Maybe adding four extra hotkeys only for this purpose while leaving the first four for whatever else the player wants. Either option has the advantage of letting you fire specific weapons you assign. Say you assign all beam turrets to hotkey X and all laser turrets to hotkey Y.

Plus if voice commands are ever added in via mod, a player would be able to shout FIRE ALL MISSILES and it would actually work :D

![](https://cdn.discordapp.com/attachments/406664114249793536/711528572300296192/unknown.png)
